### PR TITLE
feat: move `me` to `GuildMemberManager` manager

### PR DESCRIPTION
**Pull request by Syjalo**
**Please describe the changes this PR makes and why it should be merged:**
This PR moves `Guild#me` to `GuildMemberManager` manager, because it's related to GuildMembers.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

File: packages/discord.js/src/managers/GuildMemberManager.js [hunk 1]
```diff
@@ -190,6 +205,16 @@ class GuildMemberManager extends CachedManager {
...
   }
 
+  /**
+   * Fetches the owner of the guild.
+   * If the member object isn't needed, use {@link Guild#ownerId} instead.
+   * @param {BaseFetchOptions} [options] The options for fetching the member
+   * @returns {Promise<GuildMember>}
+   */
+  fetchOwner(options) {
+    return this.fetch({ ...options, user: this.ownerId });
```
Comment (muchnameless) [inline packages/discord.js/src/managers/GuildMemberManager.js:215]:
```suggestion
    return this.fetch({ ...options, user: this.guild.ownerId });
```

**Review by muchnameless — CHANGES_REQUESTED** [review]

**Comment by imranbarbhuiya** [comment]
Owner is a guild property. It's a member ik but owner of `GuildMemberManager` doesn't makes sense atleast for me.

**Comment by Syjalo** [comment]
> Owner is a guild property. It's a member ik but owner of `GuildMemberManager` doesn't makes sense atleast for me.

`GuildMemberManager` is members manager of a guild. So we fetch owner of a guild as `GuildMember` from `GuildMemberManager`.

**Comment by vladfrangu** [comment]
1. This needs a rebase
2. While I can sorta agree with moving `guild.me` to `guild.members.me`, fetchOwner should stay on the guild object imo (shortcut to guild.members.fetchGuildOwner?)

**Comment by Syjalo** [comment]
While I can sorta agree with moving `guild.me` to `guild.members.me`, fetchOwner should stay on the guild object imo (shortcut to guild.members.fetchGuildOwner?)

Why fetchOwner on GuildMembersManager? fetchOwner returns a GuildMember of a guild. All GuildMebers related thing should be in GuildMembersManager imo.
Why fetchOwner isn't called fetchGuildOwner? Because this placed on **_Guild_** Members Manager.

File: packages/discord.js/src/managers/GuildMemberManager.js [hunk 2]
```diff
@@ -190,6 +205,20 @@ class GuildMemberManager extends CachedManager {
...
   }
 
+  /**
+   * Fetches the owner of the guild.
+   * If the member object isn't needed, use {@link Guild#ownerId} instead.
+   * @param {BaseFetchOptions} [options] The options for fetching the member
+   * @returns {Promise<GuildMember>}
+   */
+  async fetchOwner(options) {
+    if (!this.ownerId) {
```
Comment (almeidx) [inline packages/discord.js/src/managers/GuildMemberManager.js:215]:
```suggestion
    if (!this.guild.ownerId) {
```

File: packages/discord.js/src/managers/GuildMemberManager.js [hunk 3]
```diff
@@ -190,6 +205,20 @@ class GuildMemberManager extends CachedManager {
...
+      throw new Error('FETCH_OWNER_ID');
+    }
+    const member = await this.members.fetch({ ...options, user: this.ownerId });
```
Comment (almeidx) [inline packages/discord.js/src/managers/GuildMemberManager.js:218]:
```suggestion
    const member = await this.fetch({ ...options, user: this.guild.ownerId });
```

File: packages/discord.js/src/managers/ThreadMemberManager.js [hunk 4]
```diff
@@ -37,6 +37,32 @@ class ThreadMemberManager extends CachedManager {
...
+
+  /**
+   * Fetches the owner of this thread. If the thread member object isn't needed,
+   * use {@link ThreadChannel#ownerId} instead.
+   * @param {BaseFetchOptions} [options] The options for fetching the member
+   * @returns {Promise<?ThreadMember>}
+   */
+  async fetchOwner({ cache = true, force = false } = {}) {
+    if (!force) {
+      const existing = this.members.cache.get(this.ownerId);
```
Comment (almeidx) [inline packages/discord.js/src/managers/ThreadMemberManager.js:57]:
```suggestion
      const existing = this.cache.get(this.thread.ownerId);
```

File: packages/discord.js/src/managers/ThreadMemberManager.js [hunk 5]
```diff
@@ -37,6 +37,32 @@ class ThreadMemberManager extends CachedManager {
...
+      if (existing) return existing;
+    }
+
+    // We cannot fetch a single thread member, as of this commit's date, Discord API responds with 405
+    const members = await this.members.fetch(cache);
```
Comment (almeidx) [inline packages/discord.js/src/managers/ThreadMemberManager.js:62]:
```suggestion
    const members = await this.fetch(cache);
```

File: packages/discord.js/src/managers/ThreadMemberManager.js [hunk 6]
```diff
@@ -37,6 +37,32 @@ class ThreadMemberManager extends CachedManager {
...
+    return members.get(this.ownerId) ?? null;
```
Comment (almeidx) [inline packages/discord.js/src/managers/ThreadMemberManager.js:63]:
```suggestion
    return members.get(this.thread.ownerId) ?? null;
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Comment by almeidx** [comment]
Missed this one: 
https://github.com/Syjalo/discord.js/blob/709ff22efd0733eff4c890bffba5893730991255/packages/discord.js/src/managers/GuildEmojiManager.js#L156

**Comment by Syjalo** [comment]
> Missed this one: https://github.com/Syjalo/discord.js/blob/709ff22efd0733eff4c890bffba5893730991255/packages/discord.js/src/managers/GuildEmojiManager.js#L156

Thanks

**Comment by splatterxl** [comment]
`owner_id` is sent by Discord in the guild object so it makes sense that it stays in the top level. Moving `.me` is fine IMO.

**Comment by Syjalo** [comment]
> `owner_id` is sent by Discord in the guild object so it makes sense that it stays in the top level. Moving `.me` is fine IMO.

It sent in guild object because the API can't sent it in members property for some reason. `GuildMemberManager` is related to specific `Guild`.

**Comment by splatterxl** [comment]
`GuildMemberManager` is a concept specific to Discord.js. The API sends an array of members in `members`, because that's what it is; a list of members.

The owner's member object is, of course, included in that list. However, `owner_id` isn't information related
to the members. It's related to the _guild_. Also, `owner_id` is still sent without the guild members intent, even when `members` isn't.
________________________________

**Comment by Syjalo** [comment]
Yes. `GuildMemberManager` is a concept specific to discord.js that manages members of a guild. Discord can't sent `guild_id` in `members` property and say us that it is the guild owner, so it sent it in guild object. A guild owner is a member so it should be in something guild members related what is the `GuildMemberManager`.

**Comment by splatterxl** [comment]
I'll put it quite simply. It's the "guild owner", not the "guild member owner".

**Comment by Syjalo** [comment]
> 

Yes it is. But all member related things of a guild are in `GuildMemberManager`.

**Comment by splatterxl** [comment]
I simply don't see how the owner of a guild is in any way related to the members?

**Comment by monbrey** [comment]
This is getting into a silly back and forth discussion when a maintainer has already made their position clear.

`GuildMemberManager#me` is fine.
`GuildMemberManager#fetchMe()` is probably fine too.

`Guild#owner_id` is where the property is, and `Guild#fetchOwner()` should exist there. You are fetching the owner of the Guild. This makes logical sense.

If you want to have a method on the manager also, it should be called `GuildMemberManager#fetchGuildOwner()`. The method has been abstracted one level away from the guild, and this should be bridged in the nomenclature.

**Comment by Syjalo** [comment]
> `GuildMemberManager#fetchGuildOwner()`. The method has been abstracted one level away from the guild, and this should be bridged in the nomenclature.

Doesn't `GuildMemberManager` already say that it related to a guild? I think it can still be abstract.

**Comment by monbrey** [comment]
> > `GuildMemberManager#fetchGuildOwner()`. The method has been abstracted one level away from the guild, and this should be bridged in the nomenclature.
> 
> Doesn't `GuildMemberManager` already say that it related to a guild? I think it can still be abstract.

But why when you can name it something less abstract, more verbose, and easier to understand?

**Comment by Syjalo** [comment]
Ok, I'll revert it later

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]
🥫

**Review by iCrawl — APPROVED** [review]

**Pull request merged**